sdk/trace/span.go

```diff
@@ -137,9 +138,14 @@ type recordingSpan struct {
 	// spanContext holds the SpanContext of this span.
 	spanContext trace.SpanContext
 
-	// attributes are capped at configured limit. When the capacity is reached
-	// an oldest entry is removed to create room for a new entry.
-	attributes *attributesMap
+	// attributes is a collection of user provided key/values. The collection
+	// is constrained by a configurable maximum. When additional attributes
+	// are added after this maximum is reached these attributes the user is
+	// attempting to add are dropped. This dropped number of attributes is
+	// tracked and reported in the ReadOnlySpan exported when the span ends.
+	attributes        map[attribute.Key]attribute.Value
+	maxAttributes     int
```

Comment:
Could we refer to `tr.provider.spanLimits.AttributeCountLimit` directly?

Reply:
SGTM, I'll push a change.